dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+        # a straight line (eq. 1)
+        # 
+        #           y_2 - y_1
+        # y - y_1 = ---------(x - x_1)
+        #           x_2 - x_1
+        # 
+        # This is rearranged to the form, ay - bx = c, where:
+        # - a = (x_2 - x_1)
+        # - b = (y_2 - y_1)
+        # - c = (x_2 - x_1)y_1 - (y_2 - y_1)x_1
+        # we can then drive the transformation to polar coordinates in 
+        # the usual way through the substitutions; y = r sin(theta), and 
+        # x = r cos(theta). The equation is now in the form 
+        #
+        #                  c
+        # r = ---------------------------
+        #     a sin(theta) - b cos(theta) 
+        #
+        a = (self.x[1:] - self.x[:-1])
+        b = (self.y[1:] - self.y[:-1])
+        c = (a * self.y[:-1] - b * self.x[:-1])
+
+        linear = c / (a * np.sin(theta) - b * np.cos(theta))
+        #return rho < (linear * self.get_pixel_scale())
+        return rho < linear
+
+
+    def _wedges(self : Layer, theta : Tensor) -> Tensor:
+        """
+        The angular bounds of each segment of an individual hexagon.
+
+        Parameters
+        ----------
+        theta : Tensor, Radians
+            The angle away from the positive x-axis of the coordinate
+            system associated with this aperture. Please note that `theta`
+            May not start at zero. 
+
+        Returns 
+        -------
+        wedges : Tensor 
+            The angular bounds associated with each pair of vertices in 
+            order. The leading axis of the Tensor steps through the 
+            wedges in order arround the circle. 
+        """
+        return (self.alpha[:-1] < theta) & (theta < self.alpha[1:])
+
+
+    def _segments(self : Layer, theta : Tensor, rho : Tensor) -> Tensor:
+        """
+        Generate the segments as a stacked tensor. 
+
+        Parameters
+        ----------
+        theta : Tensor
+            The angle of every pixel associated with the coordinate system 
+            of this aperture. 
+        rho : Tensor
+            The radial positions associated with the coordinate system 
+            of this aperture. 
+
+        Returns 
+        -------
+        segments : Tensor 
+            The bitmaps corresponding to each vertex pair in the vertices.
+            The leading dimension contains the unique segments. 
+        """
+        edges = self._edges(rho, theta)
+        wedges = self._wedges(theta)
+        return (edges & wedges).astype(float)
+        
+
+    def _aperture(self : Layer) -> Matrix:
+        """
+        Generate the BitMap representing the aperture described by the 
+        vertices. 
+
+        Returns
+        -------
+        aperture : Matrix 
+            The Bit-Map that represents the aperture. 
+        """
+        rho, theta = self._rad_coordinates()
+        segments = self._segments(theta, rho)
+        return segments.sum(axis=0)
+
+
+class CompoundAperture(eqx.Module):
```

Comment:
Yeah so as you mentioned there is already a class called `CompoundAperture` in Layers. The one in main does applies an arbitrary number of single-pixel soft-edged circular apertures and occulters. It is an aperture so it should probably live in here, but I think it should probably be re-named (using `CompoundAperture` for the one here makes a lot more sense tbh) so I think if you call if `CompoundCircularApertures` then we are good. We should keep the functionality in for now, maybe in the down the line we can deprecate it.